pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -1750,3 +1756,22 @@ type AuthorizationValidationError struct {
 func (e *AuthorizationValidationError) Error() string {
 	return e.err
 }
+
+type ScrapeClass struct {
+	// Name of the scrape class, e.g. "istio-mtls".
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Name string `json:"name"`
+
+	// Default designates a scrape class as the default class.
+	// +optional
+	Default *bool `json:"default,omitempty"`
```

Comment:
What happens if multiple scrape classes have `default: true`?

Reply:
We will fail the resource reconciliation: https://github.com/prometheus-operator/prometheus-operator/pull/6199/files#diff-453ce9c721c51bf9591c2b212e9ce50157b9c40d1b211177bf502be268099656R113